balancer/endpointsharding/endpointsharding.go

```diff
@@ -62,14 +62,31 @@ func init() {
 type ChildState struct {
 	Endpoint resolver.Endpoint
 	State    balancer.State
+
+	// Balancer exposes only the ExitIdler interface of the child LB policy.
+	// Other methods on the child policy are called only by endpointsharding.
+	Balancer balancer.ExitIdler
 }
 
 // NewBalancer returns a load balancing policy that manages homogeneous child
 // policies each owning a single endpoint.
```

Comment:
Probably a good idea to indicate that this version automatically requests reconnection for any IDLE children.

Reply:
Added a line in the doc comment.